Sources/MapboxCoreNavigation/NavigationRouter.swift

```diff
@@ -0,0 +1,337 @@
+@_implementationOnly import MapboxCommon_Private
+import MapboxDirections
+import MapboxNavigationNative
+
+/**
+ Provides alternative access to routing API.
+ 
+ Use this class instead `Directions` requests wrapper to request new routes or refresh an existing one. Depending on `RouterSource`, `NavigationRouter` will use online and/or onboard routing engines. This may be used when designing purely online or offline apps, or when you need to provide best possible service regardless of internet collection.
+ */
+public class NavigationRouter: NavigationProvider {
+    /**
+     Unique identifier for a giver request.
+     
+     Valid only for the same instance of `NavigationRouter` that issued it.
+     */
+    public typealias RequestId = UInt64
+    
+    /**
+     A request handler for the ongoing router action.
+     
+     You can use this instance to cancel ongoing task if needed. Retaining this handler will keep related `NavigationRouter` from deallocating.
+     */
+    public struct RoutingRequest: NavigationProviderRequest {
```

Comment:
Can we rename this to just Request, since it’s already nested under NavigationRouter?